SDWebImage/SDMemoryLRUCache.m

```diff
@@ -189,16 +190,19 @@ @interface SDMemoryLRUCache() {
 @implementation SDMemoryLRUCache 
 
 - (void)dealloc {
-    [_config removeObserver:self forKeyPath:NSStringFromSelector(@selector(maxMemoryCost)) context:SDMemoryCacheContext];
-    [_config removeObserver:self forKeyPath:NSStringFromSelector(@selector(maxMemoryCount)) context:SDMemoryCacheContext];
+    [_config removeObserver:self forKeyPath:NSStringFromSelector(@selector(maxMemoryCost)) context:SDMemoryLRUCacheContext];
+    [_config removeObserver:self forKeyPath:NSStringFromSelector(@selector(maxMemoryCount)) context:SDMemoryLRUCacheContext];
     
-#if SD_UIKIT
-
-    [_lru removeAll];
     pthread_mutex_destroy(&_lock);
-    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIApplicationDidReceiveMemoryWarningNotification object:nil];
+    [_lru removeAll];
```

Comment:
We should remove all instead of **JUST UIKIT** when lruCache had dealloc.

Reply:
Unter lru memory cache, I think there isn't necessary to distinguish SD_UIKIT and other platforms. But I'm not sure about it.